src/TSTransformer/nodes/class/transformDecorators.ts

```diff
@@ -6,26 +6,59 @@ import { transformPropertyName } from "TSTransformer/nodes/transformPropertyName
 import { convertToIndexableExpression } from "TSTransformer/util/convertToIndexableExpression";
 import ts from "typescript";
 
+type HasDecorators = Exclude<ts.HasDecorators, ts.AccessorDeclaration>;
+
+function canInlineDecoratorInitializer(node: HasDecorators) {
+	// more than one decorator
+	if ((ts.getDecorators(node)?.length ?? 0) > 1) {
+		return false;
+	}
+
+	// method declaration and have any parameter decorators
+	if (ts.isMethodDeclaration(node)) {
+		for (const parameter of node.parameters) {
+			if ((ts.getDecorators(parameter)?.length ?? 0) > 0) {
+				return false;
+			}
+		}
+	}
+
+	// class declaration and have any constructor parameter decorators
+	if (ts.isClassDeclaration(node)) {
+		for (const member of node.members) {
+			if (ts.isConstructorDeclaration(member)) {
+				for (const parameter of member.parameters) {
+					if ((ts.getDecorators(parameter)?.length ?? 0) > 0) {
+						return false;
+					}
+				}
+			}
+		}
+	}
+
+	return true;
+}
+
 function transformMemberDecorators(
 	state: TransformState,
-	node: ts.ClassLikeDeclaration | ts.MethodDeclaration | ts.PropertyDeclaration | ts.ParameterDeclaration,
+	node: HasDecorators,
 	callback: (expression: luau.IndexableExpression) => luau.List<luau.Statement>,
-): luau.List<luau.Statement> {
-	const result = luau.list.make<luau.Statement>();
+): [initializers: luau.List<luau.Statement>, finalizers: luau.List<luau.Statement>] {
+	const initializers = luau.list.make<luau.Statement>();
 	const finalizers = luau.list.make<luau.Statement>();
 
 	const decorators = ts.getDecorators(node);
-	const multipleDecorators = decorators !== undefined && decorators.length > 1;
+	const canInline = canInlineDecoratorInitializer(node);
 
 	for (const decorator of decorators ?? []) {
 		let [expression, prereqs] = state.capture(() => transformExpression(state, decorator.expression));
 
-		luau.list.pushList(result, prereqs);
+		luau.list.pushList(initializers, prereqs);
 
-		if (multipleDecorators && !luau.isSimple(expression)) {
+		if (!canInline && !luau.isSimple(expression)) {
```

Comment:
is `isSimple` here correct? It counts identifiers, but those might change when other decorators are initialized. Should it be `expressionMightMutate`?